### ISSUE TITLE
The garbage trucks, hearses and buses don't route properly when their target is affected by one-way road

### ISSUE THREAD

**Issue by peter-vasut** [CitiesSkylinesMods/TMPE#1600]
### Describe the problem

If the target of a vehicle is in one-way road, or road that is two-way, but has one-way road that limits how you can get to it (so it effectively behaves like one-way), the garbage trucks and hearses don't visit the area. If I put bus stop to such road, it gets placed but there is no route to it drawn.

* I've tried disabling TM:PE, after that vehicles are routed properly.
* Other vehicles (cars) visit the area apparently without problem.
* The same area works if the roads are replaced by two-way.

### Steps to reproduce

* Load the attached map.
* Observe the routing of different kind of vehicles.

**Alternatively** you can build the following network manually:

* Build big square with 4-lane one way route with bicycle path (counterclockwise direction).
* Build two-lane one-way roads going parallel from south to north with buildings in between. (I've tried Vanilla+ one with bicycle path, but also one without the bicycle lanes that's in-game to avoid possible issue caused by using roads from modpack.)
* On the left side, build road with buildings dispatching hearses and garbage trucks. I've tried to build one directly on the right side of the square and the garbage trucks despawned immediately.

### Log files

[output_log.txt](https://github.com/CitiesSkylinesMods/TMPE/files/8886852/output_log.txt)
[TMPE.log](https://github.com/CitiesSkylinesMods/TMPE/files/8886854/TMPE.log)

### Savegame?

https://steamcommunity.com/workshop/filedetails/?id=2820351225

### Screenshots?

The traffic travels from left side of screen. The area with lot's of deaths and garbage is not reached at all. The "vermont hills" area is reached in longer way, as there are one-ways around the block of buildings, that's correct behavior.
![Cities_ Skylines 12 06 2022 12_24_30](https://user-images.githubusercontent.com/8684796/173254756-3750e47a-df8f-44fb-9ff9-b59a7b21e195.png)

In this screenshot, I've placed one two-way road segment to see if it's only problem only with target being on one-way, but that does not seem to be the case, as this two-way road with one-ways around still doesn't work.
![Cities_ Skylines 12 06 2022 12_24_09](https://user-images.githubusercontent.com/8684796/173254752-d919d50d-6126-4ff6-ad34-83e75cc3ad3b.png)

**Comment by krzychu124** [comment]
Did you apply any manual lane connections or vehicle restrictions in that area?

**Comment by peter-vasut** [comment]
@krzychu124 I've used lane arrows + maybe I've tried also the lane connections previously. But currently, there should be no custom lane connections. I've also tried to remove few roads and build them from scratch, which should remove any restrictions. I didn't use any vehicle restrictions ever.

I should note that it seems like the problem occurs only when the vehicles _start_ or _terminate_ in a problematic area, as I've seen some vehicles route rough it (and they use the 4-lane road that goes around without problem to travel to different parts of the city). Placing the waste management facility to the 4-lane one-way road makes it so that it immediately despawns.

**Comment by krzychu124** [comment]
Everything works perfectly.
**You've** broke the game yourself. I've extended connections with ALL POSSIBLE options you've created. Notice missing connections marked orange with X.
Look closely, only bottom connection can exit left but **cannot** use lanes turning right after exit from roundabout.

![image](https://user-images.githubusercontent.com/19638970/173686481-dd68c768-a6ea-4350-b108-54c357c05e3a.png)

**Comment by peter-vasut** [comment]
I'm really sorry, I haven't realized that because there is only one segment between each junction and I set the lines as I did, I blocked traffic between some roads. It's quite obvious from your picture (or from the segments that get drawn on mouse over), but I didn't fully embrace what the lane tool does at the time. Placing additional buildings to troubleshoot didn't help as I only placed it on the outside roads of the block (affected by same roundabout). The private vehicles worked because they traveled from different entrance.

I can confirm, every behavior is explained, the routing works as it should.
